version: "3.10"

services:
  db:
    image: postgres:16.2
    restart: always
    volumes:
      - statistic_data:/var/lib/postgresql/data
    environment:
      POSTGRES_DB: statistic_db
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: admin
    ports:
      - "54322:5432"
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - default

  trade_marketingapp:
    build:
      dockerfile: ./Dockerfile
    container_name: trade_marketingapp
    command: >
      sh -c '/wait-for-it.sh db:5432 -- python manage.py runserver 0.0.0.0:8000'
    ports:
      - "8000:8000"
    environment:
      - DATABASE_NAME=statistic_db
      - DATABASE_USER=postgres
      - DATABASE_PASSWORD=admin
      - DATABASE_HOST=db
      - DATABASE_PORT=5432
    depends_on:
      - db
    networks:
      - default

networks:
  default:
    driver: bridge

volumes:
  statistic_data: